### DISCUSSION TITLE
Bulk move student repos to another organisation

### DISCUSSION THREAD

**Discussion by lindsaymarkward** [community/community#138336] [GitHub Education]
### Hi everyone,

Hi.
I have hundreds of student assignment repos in my staff organisation (CP1404) that I want to move to the org I usually use for student work (cp1404-students). 
I think I made a mistake when selecting the org to use when I created the GitHub Classroom assignment.
I'm not sure if it's related to my other issue with recent assignments (https://github.com/orgs/community/discussions/132182)

I'm trying to use this tool https://github.com/jdbean/github-api-bulk-transfer but getting errors.
When I try and manually move the repos, I get similar errors, so it's probably not the tool or the API. 
Here's the situation: 

**Example 1**
Private repo forked from CP1404/starter_a1_books:
In the GitHub web UI in settings, I see:
> For security reasons, you cannot change the visibility of a fork.
> This repository is not transferrable. Please contact the owner of the root repository, CP1404.

When I run the API tool for this, the error message is: 
> Repository can't be transferred and cp1404-students has insufficient collaborator seats

I can confirm that the student owner of this repo IS an "outside collaborator" in the destination org, cp1404-students.

**Example 2**
Private repo, but NOT forked:

In the UI, I DO see the option to transfer. When I try it I get the same error as the API one below.
When I run the API tool for this, the error message is (a bit different from example 1): 
> cp1404-students has insufficient collaborator seats

I can confirm that the student owner of this repo IS an "outside collaborator" in the destination org, cp1404-students.

I have hundreds of these and, a) don't want to do it manually, b) don't even know how to do it manually!
**So... what must I do?**
Thanks in advance!

**Comment by itisrude** [comment]
Hi there!

It seems like the main issue is related to the “insufficient collaborator seats” error in the `cp1404-students` org. This usually happens when the organization doesn’t have enough available seats for outside collaborators. To resolve this, you may want to check the number of seats you have available in the `cp1404-students` org and possibly upgrade your plan to allow more collaborators.

For the forked repos (Example 1), GitHub doesn’t allow changing the visibility of a fork for security reasons, and it also restricts transferring repositories that are forks. You might want to look into using the parent repo (CP1404/starter_a1_books) directly or consider deleting the fork and re-creating the repo in the `cp1404-students` org as a fresh non-forked repo.

For non-forked repos (Example 2), if you’re hitting the collaborator seat limit as well, fixing the seat availability issue should resolve both manual and API-based transfers.

Since you’re working with a large number of repos, once the collaborator seats are increased, the bulk transfer tool should work fine for non-forked repos. For the forked ones, you may need to explore alternative methods like re-creating or renaming.

Hope this helps, and good luck with the transfers!

**Reply by lindsaymarkward** [reply]
Thanks for the prompt reply!
I have never needed to change my plan - which is education - to add more student repos. There are 1.5K in the cp1404-students, so I don't think it's a plan/capacity issue, or at least, that's not the usual way I've ever had to deal with new repos. Classroom just makes them.

You wrote: 
> You might want to look into using the parent repo (CP1404/starter_a1_books) directly or consider deleting the fork and re-creating the repo

I don't know what this means, "using the parent repo". Can you help me understand?
These are finished student assignment repos that have been created in the wrong org. I don't think I can re-create them, I just need to transfer ownership so my non-student org isn't cluttered with them.

Thanks for anyone's input into how to achieve this!

**Reply by lindsaymarkward** [reply]
Hi community. Does anyone else have any suggestions for this? It can't really be an issue with my plan or “insufficient collaborator seats” as GitHub Classroom will make as many of these repos as it wants, but I can't move them.
Thanks in advance.

**Reply by lindsaymarkward** [reply]
Is there a GitHub staff member, or someone who can help with this please?

**Comment by owenarden** [comment]
I am having the exact problem described here (moving student repos from one GitHub Education organization to another). Has anyone figured out something that works?

**Comment by lindsaymarkward** [comment]
As you can see from the lack of replies, I was unable to find a solution to this problem.
I did, however, find the following excellent tool to allow me to delete many repos all at once. 
https://reporemover.xyz/
Thanks Zaahir :)
